### PR TITLE
Spinner Doc Updates

### LINKED ISSUE #2591: Spinner v2: Supporting site documentation

### PR DESCRIPTION
**Pull request by lilyvc**
closes #2591 

Added documentation for small size.

Also updated code to align with design, having `small`, `medium` and `large` sizes instead of `small`, `default` and `large`. `default` still remains and maps to `medium`, to avoid a breaking change.

### CODE REVIEW

**Comment by lilyvc** [comment]
@joshwooding is this change to the size prop okay?

cc @dplsek

File: .changeset/sixty-cameras-visit.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@salt-ds/core": minor
+---
+
+Added 'medium' size to Spinner component with 'default' size now being an alias for 'medium'
```
Comment (joshwooding) [inline .changeset/sixty-cameras-visit.md:5]: We should document an intention to deprecate and remove "default"

Comment (lilyvc) [inline .changeset/sixty-cameras-visit.md:5]:
```suggestion
Added 'medium' size to Spinner component with 'default' size now being an alias for 'medium'. 'Default' is to be deprecated.
```

File: packages/core/src/spinner/Spinner.tsx [hunk 2]
```diff
@@ -13,11 +13,23 @@ import { useDensity } from "../salt-provider";
  * Spinner component, provides an indeterminate loading indicator
  *
  * @example
- * <Spinner size="default | "large" | "small" />
+ * <Spinner size="small" | "medium" | "large" />
```
[the remark this answers is not in the document]
Comment (lilyvc) [reply packages/core/src/spinner/Spinner.tsx:16]: I think we want people to use medium so there's no need to add it to the comment

**Review by feliciackh — APPROVED** [review]

**Review by lukeac123 — APPROVED** [review]

**Pull request merged**